### PR TITLE
ENH: Add cli for CNNPatchPredictor

### PR DESCRIPTION
**Pull request by shaneahmed**
- Add command line interface for CNNPatchPredictor

### CODE REVIEW

**Review by vqdang — CHANGES_REQUESTED** [review]
Looks ok overall, im not sure why
 > file_types = tuple(file_types.split(", "))

 is replaced with this

>  file_types = string_to_tuple(file_types=file_types)

the latter is less explicit on what its trying to do.

File: tests/test_models.py [hunk 1]
```diff
@@ -938,7 +939,7 @@ def _test_predictor_output(
     pretrained_model,
     probabilities_check=None,
     predictions_check=None,
-    on_gpu=ON_GPU,
+    on_gpu=False,
 ):
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply tests/test_models.py:943]: It was not being used in previous case. you were just passing it as an argument when you were using the global variable in the function was being called. So I explicitly mentioned it.

File: tests/test_models.py [hunk 2]
```diff
@@ -949,7 +950,7 @@ def _test_predictor_output(
         inputs,
         return_probabilities=True,
         return_labels=False,
-        on_gpu=ON_GPU,
+        on_gpu=on_gpu,
     )
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply tests/test_models.py:954]: same as above, if you are using a global variable then no need to pass on the value. The function you were using was just using global value.

**Comment by shaneahmed** [comment]
> Looks ok overall, im not sure why
> 
> > file_types = tuple(file_types.split(", "))
> 
> is replaced with this
> 
> > file_types = string_to_tuple(file_types=file_types)
> 
> the latter is less explicit on what its trying to do.

The code was repeated in different cli commands. Pycharm suggests to put such code in a function as a good practice.

File: tests/test_models.py [hunk 3]
```diff
@@ -831,13 +832,13 @@ def test_wsi_predictor_api(_sample_wsi_dict):
...
-    output = predictor.predict(
+    predictor.predict(
         [_mini_wsi_svs, _mini_wsi_svs],
         masks=[_mini_wsi_msk, _mini_wsi_msk],
         mode="wsi",
         **_kwargs,
     )
-    with pytest.raises(ValueError, match=r".*save_dir.*exist.*"):
+    with pytest.raises(FileExistsError):
         _kwargs = copy.deepcopy(kwargs)
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply tests/test_models.py]: Addressed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/140/commits/ac4f89d4e06a19269983f38aaa472ea1890eb1c1

[the remark this answers is not in the document]
Comment (shaneahmed) [reply tests/test_models.py:842]: Reverted to original in this commit https://github.com/TIA-Lab/tiatoolbox/pull/140/commits/c2c0b3dded0e9b081fa595674d8c46e5202efa26

**Review by vqdang — APPROVED** [review]

**Review by simongraham — APPROVED** [review]

**Pull request merged**